### PR TITLE
Add skeleton files for changelog.

### PR DESCRIPTION
**Pull request by felicitymay**
@turbo - as discussed. This should give you a starting point for the changelog.

All the text is just placeholders.

### CODE REVIEW

File: docs/codeql/codeql-overview/codeql-changelog/index.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+CodeQL change logs
+==================
+
+TODO: Write any kind of introduction here. Any files named `codeql-cli` will appear in a bulleted list under this paragraph in descending alphabetic sort order (newest first).
+
+.. toctree::
+   :glob:
+   :reversed:
+
+   codeql-cli*
```
Comment (felicitymay) [inline docs/codeql/codeql-overview/codeql-changelog/index.rst:12]: Updating as discussed:

```suggestion
.. toctree::
   :maxdepth: 1

   codeql-cli-2.15.2
   codeql-cli-2.15.1
```

**Comment by turbo** [comment]
@felicitymay I pushed the current output from the changelog generator if you want to take a peek at the structure.

<hr>

Do no merge this PR (ever), this is meant for iteration on automatically generated changelogs.

File: docs/codeql/codeql-overview/codeql-changelog/index.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
+.. _codeql-changes:
+
+==================
+CodeQL change logs
+==================
+
+TODO: Write any kind of introduction here.
```
Comment (felicitymay) [inline docs/codeql/codeql-overview/codeql-changelog/index.rst:7]: Do we want to write an introduction here, or just delete this line? Potentially we could put here the information that's currently shown on line 12 of the individual articles?

**Review by felicitymay — COMMENTED** [review]
One thought, but otherwise the structure and layout all look good to me 👍🏻 

I know that @adityasharad was concerned about the long-term robustness of this approach, we can continue discussing our options internally.

File: docs/codeql/codeql-overview/codeql-changelog/index.rst (hunk 2 above)
Comment (turbo) [reply docs/codeql/codeql-overview/codeql-changelog/index.rst:7]: I don't have a strong opinion, happy to also add that line here. I would like to keep it in each of the docs in any case, so the info is easily available if someone is linked to a single changelog.

**Comment by felicitymay** [comment]
> Do not merge this PR (ever), this is meant for iteration on automatically generated changelogs.

Are you planning to raise a new PR against the rc branch for the upcoming 2.15.3 release, or will we retarget this one?

**Comment by turbo** [comment]
I will raise a new PR

**Comment by turbo** [comment]
Closing in favor of https://github.com/github/codeql/pull/14876

**Pull request closed without merging**